frontend/pages/hosts/HostDetailsPage/TransferHostModal/TransferHostModal.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useState } from "react";
+import { Link } from "react-router";
+import PATHS from "router/paths";
+import Modal from "components/modals/Modal";
+import Button from "components/buttons/Button";
+// ignore TS error for now until these are rewritten in ts.
+// @ts-ignore
+import Dropdown from "components/forms/fields/Dropdown";
+import { ITeam } from "interfaces/team";
+import { IHost } from "interfaces/host";
+
+interface ITransferHostModal {
+  isGlobalAdmin: boolean;
+  teams: ITeam[];
+  onSubmit: (team: ITeam) => void;
+  onCancel: () => void;
+  host: IHost;
+}
+
+interface INoTeamOption {
+  id: string;
+}
+
+const baseClass = "transfer-host-modal";
+
+const NO_TEAM_OPTION = {
+  value: "no-team",
+  label: "No team",
+};
+
+const TransferHostModal = (props: ITransferHostModal): JSX.Element => {
+  const { onCancel, onSubmit, teams, isGlobalAdmin, host } = props;
+
+  const [selectedTeam, setSelectedTeam] = useState<ITeam | INoTeamOption>();
+
+  const onChangeSelectTeam = useCallback(
+    (teamId: number | string) => {
+      if (teamId === "no-team") {
+        setSelectedTeam({ id: NO_TEAM_OPTION.value });
+      } else {
+        const teamWithId = teams.find((team) => team.id === teamId);
+        setSelectedTeam(teamWithId as ITeam);
+      }
+    },
+    [teams, setSelectedTeam]
+  );
+
+  const onSubmitTransferHost = useCallback(() => {
+    onSubmit(selectedTeam as ITeam);
+  }, [onSubmit, selectedTeam]);
+
+  const createTeamDropdownOptions = () => {
+    const teamOptions = teams.map((team) => {
+      return {
+        value: team.id,
+        label: team.name,
+      };
+    });
+    return [NO_TEAM_OPTION, ...teamOptions];
+  };
+
+  return (
+    <Modal onExit={onCancel} title={"Transfer hosts"} className={baseClass}>
+      <form className={`${baseClass}__form`}>
+        <Dropdown
+          wrapperClassName={`${baseClass}__team-dropdown-wrapper`}
+          label={"Transfer host to:"}
+          value={selectedTeam && selectedTeam.id}
+          options={createTeamDropdownOptions()}
+          onChange={onChangeSelectTeam}
+          placeholder={"Select a team"}
+          searchable={false}
+        />
+        {isGlobalAdmin ? (
```

Comment:
```suggestion
        {isGlobalAdmin && (
```

Then, you can remove the `null`